webapp/src/snippetBuilder.tsx

```diff
@@ -489,10 +516,14 @@ export class SnippetBuilder extends data.Component<SnippetBuilderProps, SnippetB
                                 {currentQuestion.errorMessage && <p className='snippet-error'>{currentQuestion.errorMessage}</p>}
                             </div>
                             {currentQuestion.hint &&
-                                <div className='snippet hint ui segment'>{pxt.Util.rlf(currentQuestion.hint)}</div>}
+                                <div className='snippet hint ui segment'>{
+                                    pxt.Util.rlf(currentQuestion.hint)
+                                        // replace new lines with <br> elements
+                                        .split("\n")
```

Comment:
is there a reason for the split/reduce vs a global string replace?

Reply:
I think so, because it's not a string I'm inserting, it's a special JSX <br> tag type.